### PR TITLE
fix(argo-cd): Clarify syntax in values.yaml

### PR DESCRIPTION
**Pull request by TheRealNoob**
the previous format was incorrectly indented and thus misleading.  Hopefully this aligns it with the other examples I see scattered around the file.

Also I should note that I think having a selector on this rule is confusing.  Rules can't have selectors.

Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [x] I have bumped the chart version according to [versioning](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#versioning)
* [x] I have updated the documentation according to [documentation](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#documentation)
* [x] I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#changelog).
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md).
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/stable/developer-guide/ci/)).

Changes are automatically published when merged to `main`. They are not published on branches.

### CODE REVIEW

**Comment by TheRealNoob** [comment]
I can bump the chart version if needed, however no functional changes were made so I lean more towards not bumping it.

**Comment by mbevc1** [comment]
Following contribution guidelines please bump the chart version and update the changelog as well.

**Comment by TheRealNoob** [comment]
> Following contribution guidelines please bump the chart version and update the changelog as well.

Done.

**Comment by jmeridth** [comment]
Need to update the artifact annotations at bottom Chart.yaml also please.

**Comment by TheRealNoob** [comment]
> Need to update the artifact annotations at bottom Chart.yaml also please.

Done.

File: charts/argo-cd/README.md [hunk 1]
```diff
@@ -105,6 +105,10 @@ For full list of changes please check ArtifactHub [changelog].
 
 Highlighted versions provide information about additional steps that should be performed by user when upgrading to newer version.
 
+### 5.23.3
```
Comment (jmeridth) [inline charts/argo-cd/README.md:108]: This is not needed.

**Review by jmeridth — CHANGES_REQUESTED** [review]

File: charts/argo-cd/README.md (hunk 1 above)
Comment (TheRealNoob) [reply charts/argo-cd/README.md:108]: That's what I thought as well when I was making the change but @mbevc1 said to update the changelog.  Did he mean something else?

Comment (TheRealNoob) [reply charts/argo-cd/README.md]: Removed.

File: charts/argo-cd/values.yaml [hunk 2]
```diff
@@ -714,7 +714,15 @@ controller:
     rules:
       # -- Deploy a PrometheusRule for the application controller
       enabled: false
-      # -- PrometheusRule.Spec for the application controller
+      # -- PrometheusRule namespace
+      namespace: "" # "monitoring"
+    
+      selector: {}
+      # prometheus: kube-prometheus
```
Comment (jmeridth) [inline charts/argo-cd/values.yaml:721]: Does this need to be indented one more to match the others in the file?

**Comment by jmeridth** [comment]
> Also I should note that I think having a selector on this rule is confusing. Rules can't have selectors.

I believe you are correct.  ServiceMonitor's have selectors, like ours [here](https://github.com/argoproj/argo-helm/blob/3a6617ce8f76284776486007cf3279501068cc83/charts/argo-cd/templates/argocd-application-controller/servicemonitor.yaml#L11-L13).  The selector was added to the PrometheusRule with #178 and especially [this commit](https://github.com/argoproj/argo-helm/pull/178/commits/539efe3a12304b87f28a745ff937a65c9e37a238).  I think it can be removed from the PromethesuRule if you'd like to remove it from the template [here](https://github.com/argoproj/argo-helm/blob/3a6617ce8f76284776486007cf3279501068cc83/charts/argo-cd/templates/argocd-application-controller/prometheusrule.yaml#L11-L13) and from your values.yaml change.  Wdyt?  This might require a deprecation notice in README.md. /cc @mkilchhofer wdyt?

File: charts/argo-cd/values.yaml (hunk 2 above)
Comment (TheRealNoob) [reply charts/argo-cd/values.yaml:721]: most of the helm charts that I've looked at (public and private) treat the `#` comment symbol as invisible.  Expecting the consumer to replace the symbol with a space.  Doing it the other way around can lead to confusion in some scenarios.  An example:
```yaml
rootDict:
  key1: value1
# key2: value2
```

With this in mind I see that a lot of the helm chart does not treat the comment as invisible, however there are some exceptions.  [Comment followed by two spaces](https://github.com/argoproj/argo-helm/blob/3a6617ce8f76284776486007cf3279501068cc83/charts/argo-cd/values.yaml#L85) and [Invisible comment](https://github.com/argoproj/argo-helm/blob/3a6617ce8f76284776486007cf3279501068cc83/charts/argo-cd/values.yaml#L91).  This is by no means comprehensive I'm just cherry picking.  The way that I'm used to reading values files tells me that [this ](https://github.com/argoproj/argo-helm/blob/3a6617ce8f76284776486007cf3279501068cc83/charts/argo-cd/values.yaml#L702)is meant to be indented by 4 spaces, however I know that's not the case.  You tell me which way you'd like it formatted and I'll do that.

**Comment by TheRealNoob** [comment]
I agree `.Values.controller.metrics.rules.selector` should be removed, however I don't think this is the MR to do it in.  I think that change should probably bump the chart version from `5.23.2` to `5.24.0` just because it's making a minor breaking change.  I apologize if you already know the rest of this comment, I just thought I'd explain my thought process now while I have the time to do it.

Helm versioning follows the SemVer scheme which defines the naming convention as `MAJOR.MINOR.PATCH` which says you increment the:
* MAJOR version when you make incompatible API changes
* MINOR version when you add functionality in a backwards compatible manner
* PATCH version when you make backwards compatible bug fixes
[source](https://semver.org/)

Removing the selector is a breaking change, however I really don't feel it justifies a major number.  So I look at the above more as guidelines than rules :).  I think this current MR that I'm doing here definitely falls under the `patch` category.

**Comment by mkilchhofer** [comment]
When I first looked at this, when I started to contribute and help maintain here, I was also confused.

The idea behind this "selector" is for the prometheus-operator running on the cluster. You can configure on your `kind: Prometheus` what ServiceMonitors/PrometheusRules/etc should be picked up. Eg if you have multiple Prometheus instances running like one instance for the core cluster and others for Applications or Tenants (when users do soft tenancy on kubernetes).

Eg. when you use the kube-prometheus-stack chart, the selector by default is `release: <ReleaseName`. Some users (the one sonly using one Prom instance per cluster) disable this feature on Prometheus operator completely by setting `serviceMonitorSelectorNilUsesHelmValues: false`

Does that help?

File: charts/argo-cd/values.yaml (hunk 2 above)
Comment (TheRealNoob) [reply charts/argo-cd/values.yaml:721]: let me know which way you'd like it formatted

Comment (jmeridth) [reply charts/argo-cd/values.yaml:721]: Whether it's right or wrong, my vote is for consistency with the rest of this project's current choice. Wdyt?

Comment (TheRealNoob) [reply charts/argo-cd/values.yaml]: alright indented to match the others.

**Comment by TheRealNoob** [comment]
@jmeridth ready for review

**Review by jmeridth — APPROVED** [review]

**Comment by mbevc1** [comment]
You need to update to docs as well and that will pass the tests.

**Comment by TheRealNoob** [comment]
> You need to update to docs as well and that will pass the tests.

@mbevc1 Please point me to what specifically needs to be updated.  I added a node to the changelog as you instructed me before and @jmeridth told me to remove it.  I agree with him that that changelog (at present) only contains info about changes which could impact users of this helm chart.

**Comment by jmeridth** [comment]
@TheRealNoob run `./scripts/helm-docs.sh` locally to have it update README.md for you. Should fix the issue.

**Comment by mbevc1** [comment]
@TheRealNoob what @jmeridth said :smile:

**Review by mkilchhofer — APPROVED** [review]

**Review by jmeridth — APPROVED** [review]

**Pull request merged**